### ISSUE TITLE
java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.GenericMutableRow cannot be cast to org.apache.spark.sql.Row

### ISSUE THREAD

**Issue by bagonzalo** [harsha2010/magellan#62]
Hi,

I receive the following error when trying to read a .shp folder:

java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.GenericMutableRow cannot be cast to org.apache.spark.sql.Row

I read that it is related with magellan package trying to use schema "GenericMutableRow", which is no longer supported in recent versions of Spark. Is it the case? Any prospects of updating it?

Thank you

**Comment by harsha2010** [comment]
yes, there is a branch i have been working on to support spark 1.6. Its ready for a final pass in the next day or two and I'll cut a release 1.0.4 with spark 1.6 support by this weekend
